### PR TITLE
chore: flatten core files structure

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Flattens the files in the core package before future refactorings

### Why is it needed?

Simplification

### How to test it?

make sure all the tests are passing

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/core/core/src/domain/content-type/index.ts [hunk 1]
```diff
@@ -133,6 +133,7 @@ const pickSchema = (model: Schema.ContentType) => {
       'options',
       'pluginOptions',
       'attributes',
+      'kind',
```
Comment (Convly) [inline packages/core/core/src/domain/content-type/index.ts:136]: Why is this part of this PR? Was this missing?

File: packages/core/core/src/loaders/plugins/index.ts [hunk 2]
```diff
@@ -3,8 +3,8 @@ import fse from 'fs-extra';
 import { defaultsDeep, defaults, getOr, get } from 'lodash/fp';
 import { env } from '@strapi/utils';
 import type { Strapi, Plugin } from '@strapi/types';
-import { loadFile } from '../../app-configuration/load-config-file';
-import loadFiles from '../../../load/load-files';
+import { loadFile } from '../../utils/load-config-file';
```
Comment (Convly) [inline packages/core/core/src/loaders/plugins/index.ts:6]: Why not taking this opportunity to rename this `loadConfigFile` to match the file name (makes it less confusing regarding the export just below)

Comment (alexandrebodin) [reply packages/core/core/src/loaders/plugins/index.ts:6]: sure I can 👍

File: packages/core/core/src/domain/content-type/index.ts (hunk 1 above)
Comment (alexandrebodin) [reply packages/core/core/src/domain/content-type/index.ts:136]: I was getting a type error I'll recheck

Comment (alexandrebodin) [reply packages/core/core/src/domain/content-type/index.ts:136]: Yeah I need that fix

**Review by Marc-Roig — APPROVED** [review]

File: packages/core/core/src/loaders/plugins/index.ts (hunk 2 above)
Comment (alexandrebodin) [reply packages/core/core/src/loaders/plugins/index.ts]: Done

**Review by Convly — APPROVED** [review]

File: packages/core/core/src/Strapi.ts [hunk 3]
```diff
@@ -30,14 +30,14 @@ import type {
   Schema,
 } from '@strapi/types';
 
-import loadConfiguration from './core/app-configuration';
+import loadConfiguration from './configuration';
```
Comment (innerdvations) [inline packages/core/core/src/Strapi.ts:33]: Should we take the opportunity to remove the default export here and make it a named export? That way it won't matter that the filename is different from what we're calling the import.

**Review by innerdvations — APPROVED** [review]

File: packages/core/core/src/Strapi.ts (hunk 3 above)
Comment (alexandrebodin) [reply packages/core/core/src/Strapi.ts:33]: I'll enable the no-default rule later and do a oneshot pass most likely

**Pull request merged**